### PR TITLE
Set package long description encoding to UTF-8

### LINKED ISSUE #2110: Feast setup script returns "'charmap' codec can't decode byte" on Windows
## Expected Behavior

`pip install -e "sdk/python[ci]"` must succeed and install feast package in editable mode.

## Current Behavior

Installation error:

```
Obtaining file:///C:/Users/<username>/Downloads/feast/sdk/python
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\<username>\Downloads\feast\venv\Scripts\python.exe' 'C:\Users\<username>\Downloads\feast\venv\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' get_requires_for_build_wheel 'C:\Users\<folder_name>\AppData\Local\Temp\tmpwd6unq88'
       cwd: C:\Users\<username>\Downloads\feast\sdk\python
  Complete output (20 lines):
  Traceback (most recent call last):
    File "C:\Users\<username>\Downloads\feast\venv\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 280, in <module>
      main()
    File "C:\Users\<username>\Downloads\feast\venv\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 263, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "C:\Users\<username>\Downloads\feast\venv\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 114, in get_requires_for_build_wheel
      return hook(config_settings)
    File "C:\Users\<username>\AppData\Local\Temp\pip-build-env-2u3vtd1l\overlay\Lib\site-packages\setuptools\build_meta.py", line 162, in get_requires_for_build_wheel
      return self._get_build_requires(
    File "C:\Users\<username>\AppData\Local\Temp\pip-build-env-2u3vtd1l\overlay\Lib\site-packages\setuptools\build_meta.py", line 143, in _get_build_requires
      self.run_setup()
    File "C:\Users\<username>\AppData\Local\Temp\pip-build-env-2u3vtd1l\overlay\Lib\site-packages\setuptools\build_meta.py", line 267, in run_setup       
      super(_BuildMetaLegacyBackend,
    File "C:\Users\<username>\AppData\Local\Temp\pip-build-env-2u3vtd1l\overlay\Lib\site-packages\setuptools\build_meta.py", line 158, in run_setup       
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 137, in <module>
      LONG_DESCRIPTION = f.read()
    File "c:\users\<username>\appdata\local\programs\python\python39\lib\encodings\cp1252.py", line 23, in decode
      return codecs.charmap_decode(input,self.errors,decoding_table)[0]
  UnicodeDecodeError: 'charmap' codec can't decode byte 0x90 in position 1599: character maps to <undefined>
```

## Steps to reproduce

Install feast for development:

```
pip install -e "sdk/python[ci]"
```

### Specifications

- Version: Master branch
- Platform: Windows 10
- Python version: 3.9.5
- pip version: 21.3.1
- virtualenv version: 20.10.0

## Possible Solution

`README.md` contains emojis resulting character encoding issues in the package long description. This PR sets package long description encoding to UTF-8.

```python
# README file from Feast repo root directory
README_FILE = os.path.join(repo_root, "README.md")
with open(README_FILE, "r", encoding="utf8") as f:
    LONG_DESCRIPTION = f.read()
```

Pull request #2111

### PR DESCRIPTION
**Pull request by danilopeixoto**
**What this PR does / why we need it**:

Fix feast setup script for Windows.
`README.md` contains emojis resulting character encoding issues in the package long description. This PR sets package long description encoding to UTF-8.

**Which issue(s) this PR fixes**:

Fixes #2110

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/setup.py [hunk 1]
```diff
@@ -117,16 +117,23 @@
     "pre-commit",
     "assertpy==1.1",
     "pip-tools",
-] + GCP_REQUIRED + REDIS_REQUIRED + AWS_REQUIRED
+    *GCP_REQUIRED,
```
Comment (pyalex) [inline sdk/python/setup.py:120]: Hi @danilopeixoto, what purpose of this change? isn't `a + b + c` syntax easier to read?

File: sdk/python/setup.py [hunk 2]
```diff
@@ -117,16 +117,23 @@
...
+    *CI_REQUIRED,
+]
 
 # Get git repo root directory
 repo_root = str(pathlib.Path(__file__).resolve().parent.parent.parent)
 
 # README file from Feast repo root directory
 README_FILE = os.path.join(repo_root, "README.md")
-with open(README_FILE, "r") as f:
+with open(README_FILE, "r", encoding="utf8") as f:
```
Comment (pyalex) [inline sdk/python/setup.py:136]: Nit: I believe the correct name is `utf-8`

Comment (danilopeixoto) [reply sdk/python/setup.py:136]: `utf-8` was my first option before I look for references and usage in other source files and they was using `utf8`:

https://github.com/feast-dev/feast/blob/7538f7e643765b5abcf004e3cea2918928c8f0dc/sdk/python/feast/infra/key_encoding_utils.py#L34

`utf8` is valid according the aliases column in the [Standard Encodings](https://docs.python.org/3/library/codecs.html#standard-encodings) table.

Do you think it's better to switch to `utf-8`?

Comment (pyalex) [reply sdk/python/setup.py:136]: I don't have a strong preference. If that's a standardized alias, let's keep it.

File: sdk/python/setup.py (hunk 1 above)
Comment (danilopeixoto) [reply sdk/python/setup.py:120]: These changes were really aesthetic. I can reverse those changes if it's an issue (not clear to read).

I changed due to symmetry. Eg:

```python
CI_REQUIRED = [
    "cryptography==3.3.2",
    "flake8",
    ...,
    *GCP_REQUIRED,
    *REDIS_REQUIRED,
    *AWS_REQUIRED,
]
```
vs

```python
CI_REQUIRED = [
    "cryptography==3.3.2",
    "flake8",
    ...,
] + GCP_REQUIRED + REDIS_REQUIRED + AWS_REQUIRED
```

Comment (pyalex) [reply sdk/python/setup.py:120]: Yes, can you revert this please? @danilopeixoto

Comment (danilopeixoto) [reply sdk/python/setup.py]: Yes! Reverted to plus sign.

**Review by pyalex — APPROVED** [review]

**Comment by pyalex** [comment]
@danilopeixoto can you please properly sign your commits? (see the failing DCO check)

File: sdk/python/setup.py [hunk 3]
```diff
@@ -119,14 +119,17 @@
     "pip-tools",
 ] + GCP_REQUIRED + REDIS_REQUIRED + AWS_REQUIRED
 
-DEV_REQUIRED = ["mypy-protobuf==1.*", "grpcio-testing==1.*"] + CI_REQUIRED
+DEV_REQUIRED = [
```
Comment (pyalex) [inline sdk/python/setup.py:122]: apparently this change doesn't pass python lint, @danilopeixoto

Comment (pyalex) [reply sdk/python/setup.py:122]: Can you please revert it as well?

Comment (danilopeixoto) [reply sdk/python/setup.py]: Removing newlines...

**Review by achals — APPROVED** [review]

**Pull request merged**